expo/config-plugins
# Herytz/react native google signin

# Why

[react-native-google-signin](https://github.com/react-native-google-signin/google-signin) is not compatible with Prebuild. This plugin allows you to configure react-native-google-signin using Prebuild.

# How

I take a look at the other Expo Config Plugins in this repo and copying their implementation.

# Test Plan

I tested it locally

1. Install react-native-google-signin
2. Add configuration
```json
{
  "expo": {
    "plugins": [
      [
        "@config-plugins/react-native-google-signin",
        {
          "serverClientId": "my_serverClientId",
          "iosClientId": "my_iosClientId",
          "iosUrlScheme": "my_iosUrlScheme"
        }
      ]
    ]
  }
}
```
4. Check Info.plist: `ios/[project_id]/Info.plist`
```
...

<key>CFBundleURLTypes</key>
    <array>

      ...

      <dict>
        <key>CFBundleURLSchemes</key>
        <array>
          <string>my_iosUrlScheme</string>
        </array>
      </dict>
    </array>

...

<key>GIDClientID</key>
<string>my_iosClientId</string>

...
```
5. Check strings.xml: `android/app/src/main/res/values/strings.xml`
```
<resources>
  ...
  <string name="server_client_id">my_serverClientId</string>
</resources>
```
7. Run `npx expo prebuild --clean`
